packages/headless/src/features/search-parameters/search-parameter-actions.ts

```diff
@@ -1,12 +1,22 @@
 import {createAction} from '@reduxjs/toolkit';
-import {
-  EnableQuerySyntaxParam,
-  QueryParam,
-} from '../../api/search/search-api-params';
+import {SearchRequest} from '../../api/search/search/search-request';
 import {validatePayloadSchema} from '../../utils/validate-payload';
 import {searchParametersDefinition} from './search-parameter-schema';
 
-export type SearchParameters = QueryParam & EnableQuerySyntaxParam;
+export type SearchParameters = Omit<
+  SearchRequest,
+  | 'organizationId'
+  | 'accessToken'
+  | 'visitorId'
+  | 'url'
+  | 'enableDidYouMean'
+  | 'fieldsToInclude'
+  | 'facetOptions'
+  | 'facets'
+  | 'searchHub'
+  | 'pipeline'
+  | 'context'
```

Comment:
I want to use this opportunity to revisit the discussion from the design doc.

The keys above are currently not supported.
I think we're in agreement that keys from `organizationId` until `facetOptions` inclusive should **not** be supported.

The `facets` key will be gradually included in separate PRs.

For the last three, `searchHub`, `pipeline` and `context`, I am less sure that they should be excluded. The goal of the url feature is to help two users share results, and these three params are often used to set conditions in the admin.

If there is still uncertainty, we can also keep things as they are and revisit later,

Reply:
> Two users share results

Exactly ! Remember that we're talking about two "end users" here.

Which means "parameters that the end user actually has control over, or can modify due to a widget/component/control" in the UI.

We expose `q`, because it typically comes from a search box, which the end users has control over.
We expose `firstResult`/`numberOfResults` because it typically comes from a pager and a results per page component.
Same for facets.

When was the last time you saw a Coveo user interface, where, as the end user on some public website, you had a drop down to select the pipeline ? Or the search hub ? Or to control the "context" ?

It would be pretty silly. Because those concepts don't mean anything for the end user.

These are typically controlled by code, written by the developer using headless, and that the end user has no control over.


An example that might help illustrate what I'm trying to say: 

When you go on youtube, and I share to you `https://www.youtube.com/results?search_query=coveo`, do you think we will see the exact same results ? The URL does not contain my whole google account "context", so that we see the exact same thing.

If you open the "filter" button on that UI, you will see that the URL will get modified, according to the filter selection. Because, for those, you have control over as an end user. 

But, still, you can be sure that there's a ton of code behind the scene that tailor what you and me will see, which will be different.

Also in the design doc, something important was the fact that all of this still needs to be configurable (eventually/later version). Because the "default" parameter we want to expose in the URL will change from app to app.

Because, even though it would be rare, it's "possible" that somebody would actually want to create an app with a dropdown that allow you to select your pipeline, for some reason. 
And then, in that case, it means that parameter would need to be exposed in the URL.

Reply:
Also, think of the end result if we start to serialize everything in the URL.

It would become very, very large.

Even facet are kind of a problem because it's a lot of information.

Reply:
A million different users, a million experiences. 😂

Reply:
**Pipeline use-case**

> When was the last time you saw a Coveo user interface, where, as the end user on some public website, you had a drop down to select the pipeline ? Or the search hub ? Or to control the "context" ?

My main concern with this statement is someone could set up headless to adjust these parameters indirectly in response to user-interaction. e.g.

A search page has two search boxes. Depending on which they use, the pipeline will be adjusted. The user then shares a url which contains their query, but not the pipeline that was programmatically set. Clicking the url will not show the same results.

**Relaxing the Same-Result Assumption**

With that said, I see your point with the youtube example and I think it is fair to relax the assumption of "Two users seeing the same results", and so would not need to include the last three parameters. I guess ultimately there are two ways to share:

1. Sharing search page urls. Results may vary for two users.
2. Sharing a url to a specific document/video. Both users will see the same document assuming they have the same permissions.

**Configuring additional params**

I will open a jira to look into how to best do this.